internal/pkg/manifest/pipeline.go

```diff
@@ -205,6 +207,15 @@ type Deployment struct {
 	DependsOn      []string `yaml:"depends_on"`
 }
 
+// PrePostDeployments represent a directed graph of cloudformation deployments.
+type PrePostDeployments map[string]*PrePostDeployment
+
+// PrePostDeployment is the config for a pre- or post-deployment action.
```

Comment:
nit: this would give us some context on why "buildspec" is here!
```suggestion
// PrePostDeployment is the config for a pre- or post-deployment action backed by CodeBuild.
```